test/services/schema-file-updater.test.js

```diff
@@ -21,7 +21,7 @@ describe('services > schema-file-updater', () => {
   const schemaSerializer = new SchemaSerializer();
   const { options: serializerOptions } = schemaSerializer;
   const buildSchema = (collection, metas = meta) =>
-    new SchemaFileUpdater('test.json', collection, metas, serializerOptions).perform();
+    new SchemaFileUpdater({ logger }).update('test.json', collection, metas, serializerOptions);
 
   // NOTICE: Expecting `fs.writeFileSync` second parameter to be valid JSON.
   it('should call fs.writeFileSync with a valid JSON as data', () => {
```

Comment:
[INFORMATION TO KNOW BUT WORKLOAD TO CHANGE]
This advice means a rewrite of this file, ok you can let it as is if you want, the effort to make it square doesn't bring enough, but it's important to not continue this way in future files.

This is an integrated test. When testing services, a unit test is preferable.
Unit test means testing one method only, mocking all dependencies, and to have all dependencies of the tested method injected. To test each method independently (excluding even methods of the same class)

A real integrated test of this file is when testing with the caller (method `generateAndSendSchema` in `index.js`, which is in fact the init method of forest-express).

Reply:
[IMPORTANT]
You can consider create `writeFileSync` in the context.
You can do `.addInstance('writeFileSync', (args) => fs.writeFileSync(...args))`
And then injecting it in your service.

Like that in test, you will mock the write (by adding `jest.fn()`) so you can test call without a spy, and without really writing to the file which is consuming.

Then people can reuse the write-file-to-disk service your created, and tests are always the same.

Reply:
Thank you.

I refactored to follow the `[IMPORTANT]` here: https://github.com/ForestAdmin/forest-express/pull/528/commits/2ce6ff049b8ef3cf0962a18e478e88995a518a60

I had to ignore `[INFORMATION TO KNOW BUT WORKLOAD TO CHANGE]` because it's cumbersome.